lib/membrane_h264_ffmpeg/encoder.ex

```diff
@@ -123,7 +123,8 @@ defmodule Membrane.H264.FFmpeg.Encoder do
   def handle_caps(:input, %Raw{} = caps, _ctx, state) do
     {framerate_num, framerate_denom} = caps.framerate
 
-    with {:ok, encoder_ref} <-
+    with {:ok, buffers} = flush_encoder_if_exists(state),
```

Comment:
Using `=` in `with` will result in raising an error in case of not matching pattern
```suggestion
    with {:ok, buffers} <- flush_encoder_if_exists(state),
```